scripts/add-pool.js

```diff
@@ -8,152 +8,242 @@ require('dotenv').config()
-    const pool = {
-      name,
-      poolName,
-      address: web3.utils.toChecksumAddress(address),
-      asset: handleAssetName(asset),
-      birthblock,
-      chainId: Number.parseInt(chainId),
-      riskLevel:
-        poolName.startsWith('va') ||
-        poolName.startsWith('ve') ||
-        chainId !== '1'
-          ? 4
-          : 3,
-      stage: 'alpha',
-      version,
-      symbol,
-      decimals,
-      logoURI: `https://raw.githubusercontent.com/vesperfi/metadata/master/src/logos/${imageName}.svg`,
-      type: poolName.startsWith('ve') ? 'earn' : 'grow'
+    function unwrapped(asset) {
+      return {
+        'WAVAX:43114': 'AVAX',
+        'WETH:1': 'ETH',
+        'WMATIC:137': 'MATIC',
+        'WBNB:56': 'BNB'
+      }[`${asset}:${chainId}`]
     }
 
-    metadata.pools.push(pool)
-    metadata.pools
-      .sort((a, b) => b.birthblock - a.birthblock)
-      .sort((a, b) => a.chainId - b.chainId)
-    fs.writeFileSync(
-      'src/vesper-metadata.json',
-      JSON.stringify(metadata, null, 2)
-    )
+    if (
+      metadata.pools.find(
+        pool =>
+          pool.address.toLowerCase() === address.toLowerCase() &&
+          pool.chainId === Number.parseInt(chainId)
+      )
+    ) {
+      console.log('Pool already added')
+      process.exit(0)
+    }
 
-    console.log(`${name} added!`)
-  })
-  .catch(function (err) {
-    console.error(`Failed adding pool: ${err.message}`)
+    // @ts-ignore ts(2351)
+    const web3 = new Web3(getNodeUrl(chainId))
+    const contract = new web3.eth.Contract(poolAbi, address)
+
+    Promise.all([
+      contract.methods.name().call(),
+      contract.methods.token().call(),
+      contract.methods.token().call().then(getTokenSymbol),
+      getBirthBlock(),
+      contract.methods.VERSION().call().then(getMayorVersion),
+      contract.methods.symbol().call(),
+      contract.methods.decimals().call().then(Number.parseInt)
+    ])
+      .then(function ([
+        poolName,
+        token,
+        asset,
+        birthblock,
+        version,
+        symbol,
+        decimals
+      ]) {
+        const name = poolName.split(' ')[0]
+        const imageName = name.toLowerCase().split('.')[0]
+        const pool = {
+          name,
+          poolName,
+          address: web3.utils.toChecksumAddress(address),
+          asset: handleAssetName(asset),
+          birthblock,
+          chainId: Number.parseInt(chainId),
+          riskLevel:
+            poolName.startsWith('va') ||
+            poolName.startsWith('ve') ||
+            chainId !== '1'
+              ? 4
+              : 3,
+          stage: 'alpha',
+          version,
+          symbol,
+          decimals,
+          logoURI: `https://raw.githubusercontent.com/vesperfi/metadata/master/src/logos/${imageName}.svg`,
+          type: poolName.startsWith('ve') ? 'earn' : 'grow',
+          collateral: {
+            address: token,
+            decimals,
```

Comment:
`decimals` are the pool token decimals, not the collateral token decimals. Should make a call similar to `getTokenSymbol` to get the right value.

Also, should check the data inserted in the JSON for possible errors (WBTC is one of the collateral tokerns not having 18 decimals).

Reply:
ufff, good catch! I'm not sure why I was thinking that the decimals property belonged to the collateral instead of the pool 🤔. Even more, I was afirming that on the pr description😂😅. Solved.